File: 1629.py
from collections import deque

n = int(input())
movies_times = []
for i in range(n):
    a, b = map(int, input().split())
    movies_times.append((a, b))

movies_times.sort(key=lambda v: v[1])
current_movie = None
curr_v = 0

for movie_time in movies_times:
    a, b = movie_time
    if current_movie == None or current_movie[1] <= a or current_movie[0] >= b:
        curr_v += 1
        current_movie = movie_time

print(curr_v)

File: 1629.cpp
#include <iostream>
#include <vector>
#include <algorithm>

using t_movie = std::pair<unsigned int, unsigned int>;

int main(){
    int n;
    std::cin >> n;

    std::vector<t_movie> movies (n);

    for (int i = 0; i < n; i++){
        std::cin >> movies[i].first >> movies[i].second;
    }

    std::sort(movies.begin(), movies.end(), [](t_movie & l, t_movie & r){
        return l.first <= r.first && (l.second - l.first) <= (r.second - r.first);
    });

    int curr_v = 0;
    t_movie current_movie {0, 0};

    for (auto movie : movies){
        if (current_movie.first == 0 || current_movie.second <= movie.first || current_movie.first >= movie.second){
            curr_v ++;
            current_movie = movie;
        }
    }

    std::cout << curr_v << std::endl;
}